src/test/fuzz/util.h

```diff
@@ -512,4 +512,63 @@ void ReadFromStream(FuzzedDataProvider& fuzzed_data_provider, Stream& stream) no
     }
 }
 
+class FuzzedSocket : public MockableSocket
+{
+    FuzzedDataProvider& m_fuzzed_data_provider;
+
+public:
+    explicit FuzzedSocket(FuzzedDataProvider& fuzzed_data_provider) : m_fuzzed_data_provider{fuzzed_data_provider}
+    {
+    }
+
+    ssize_t send(const char* buf, size_t len, int flags) override
+    {
+        if (m_fuzzed_data_provider.ConsumeBool()) {
+            return len;
+        }
+        return m_fuzzed_data_provider.ConsumeIntegralInRange<ssize_t>(-1, len);
```

Comment:
Just a note: this could return `0` from `send()`. I am not sure if [`send(2)`](https://linux.die.net/man/2/send) could ever [return `0`](https://stackoverflow.com/questions/3081952/with-c-tcp-sockets-can-send-return-zero#3082032):

> On success, these calls return the number of characters sent. On error, -1 is returned

anyway, I think it is ok to return `0` - to check that our code does not get bricked by that (in case `send(2)` actually returns `0`).